packages/box/src/dfuse-receiver.ts

```diff
@@ -0,0 +1,242 @@
+        interval: 30,
+    };
+
+    dfuseClient = createDfuseClient({
+        apiKey: dfuseConfig.apiKey,
+        network: dfuseConfig.apiNetwork,
+        authUrl: dfuseConfig.authNetwork,
+        httpClientOptions: {
+            fetch: nodeFetch,
+        },
+        graphqlStreamClientOptions: {
+            socketOptions: {
+                webSocketFactory: (url) =>
+                    webSocketFactory(url, ["graphql-ws"]),
+            },
+        },
+        streamClientOptions: {
+            socketOptions: {
+                webSocketFactory,
+            },
+        },
+    });
+
+    constructor(storage: Storage) {
+        this.storage = storage;
+    }
+
+    pushTrx(trx: JsonTrx) {
+        if (this.unpushedTransactions.length) {
+            const prev = this.unpushedTransactions[
+                this.unpushedTransactions.length - 1
+            ];
+            if (trx.undo != prev.undo || trx.block.id != prev.block.id) {
+                if (prev.undo) this.storage.undo(prev.block.id);
+                else if (prev.trace) {
+                    const block = { ...prev.block, transactions: [] as any[] };
+                    for (let t of this.unpushedTransactions) {
+                        block.transactions.push({
+                            id: t.trace.id,
+                            actions: t.trace.matchingActions.map((a) => ({
+                                seq: a.seq,
+                                firstReceiver: a.account,
+                                receiver: a.receiver,
+                                name: a.name,
+                                hexData: a.hexData,
+                            })),
+                        });
+                    }
+                    this.storage.pushJsonBlock(
+                        JSON.stringify(block),
+                        prev.irreversibleBlockNum
+                    );
+                }
+                this.unpushedTransactions = [];
+            }
+        }
+        if (trx.trace !== null) this.unpushedTransactions.push(trx);
+    }
+
+    onMessage(message: GraphqlStreamMessage<any>, stream: Stream): void {
+        if (message.type === "data") {
+            const trx: JsonTrx = message.data.searchTransactionsForward;
+            const prev =
+                this.jsonTransactions.length > 0
+                    ? this.jsonTransactions[this.jsonTransactions.length - 1]
+                    : null;
+            logger.info(
+                `${trx.undo ? "undo block" : "recv block"} ${trx.block.num} ${
+                    trx.trace
+                        ? "trx " + trx.trace.id
+                        : "no matching transactions"
+                }`
+            );
+            if (trx.trace || (prev && prev.trace)) {
+                this.jsonTransactions.push(trx);
+                this.pushTrx(trx);
+                if (
+                    this.jsonTransactions.length - this.numSaved > 10 ||
+                    !trx.trace
```

Comment:
what a new trx without a trace means?

Reply:
It's not a trx; instead it's an indication that `interval` blocks have passed without any matching transactions